test/Dialect/FIRRTL/parse-basic.fir

```diff
@@ -1471,3 +1471,18 @@ circuit Probes_refexprs:
     force_initial(rwprobe(`test`), `test`)
     ; CHECK: force_initial
     force_initial(`9`.`0`.rwprobe, `test`.`0`)
+
+// -----
+FIRRTL version 3.0.0
+circuit Top:
+  type WordType = UInt<32>
+  type ValidType = UInt<1>
+  type Data = {w: WordType, valid: ValidType, flip ready: UInt<1>}
+  type Complex = { real: SInt<10>, imag: SInt<10> }
+  type ProbeComplex = Probe<Complex>
```

Comment:
Maybe check vectors parse appropriately too, and/or interaction with const?

A few error cases are warranted as well, please.